src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -500,7 +509,7 @@ event::ptr primitive_inst::realloc_if_needed() {
     const auto& users = get_user_insts();
     if (users.size() == 1 && users.front()->get_node().is_type<concatenation>()) {
         auto concat_inst = users.front();
-        if (concat_inst->can_be_optimized()) {
+        if (concat_inst->can_be_optimized() && !has_subgraph_dependency(concat_inst->dependencies())) {
```

Comment:
I think `concat_inst->get_node().is_in_shape_of_subgraph()` should work

Reply:
I've tried it before and it didn't work

Reply:
OK, then I still suspect an issue with shape flow markup. Let's keep it as is for now, and follow up this later.